src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -2303,6 +2303,35 @@ private static BigInteger Subtract(uint[]? leftBits, int leftSign, uint[]? right
             return !right.Equals(left);
         }
 
+        /// <summary>
+        /// Gets the number of bits required to represent this BigInteger instance in two's complement notation without the sign bit.
+        /// </summary>
+        /// <returns>The non-negative number of bits in two's complement notation without the sign bit.</returns>
+        /// <remarks>If <see cref="IsZero"/> is true, this method returns 0. For positive integers the return value is equal to the binary representation string length.</remarks>
+        public long GetBitLength()
+        {
+            AssertValid();
+
+            uint highValue;
+            int byteLength;
+            uint[]? bits = _bits;
+
+            if (bits == null)
+            {
+                byteLength = 1;
+
+                int sign = _sign;
+                highValue = sign < 0 ? (uint)-sign : (uint)sign;
+            }
+            else
+            {
+                byteLength = bits.Length;
+                highValue = bits[byteLength - 1];
+            }
+
+            return byteLength * 32L - BitOperations.LeadingZeroCount(highValue);
```

Comment:
The `byteLength` name seems misleading because it counts 32-bit units and `System.Byte` is 8-bit.